plugins/Goals/vue/src/ManageGoals/ManageGoals.vue

```diff
@@ -675,6 +686,49 @@ export default defineComponent({
         this.isLoading = false;
       });
     },
+    storeNotification(goalId:string|number, isCreate:boolean) {
+      try {
+        sessionStorage.setItem(notificationKey, JSON.stringify({ goal: goalId, create: isCreate }));
+      } catch (e) {
+        // Do nothing
+      }
+    },
+    getStoredNotification() {
+      const pendingNotification = sessionStorage.getItem(notificationKey);
+      if (pendingNotification) {
+        sessionStorage.removeItem(notificationKey);
+        try {
+          let { goal, create } = JSON.parse(pendingNotification);
+          if (goal) {
+            goal = parseInt(goal, 10); // we make sure this is an int
+          }
+          create = !!create; // we make sure this is a boolean
+          return { goal, create };
+        } catch (e) {
+          return null;
+        }
+      }
+      return null;
+    },
+    showNotificationMessage(goalId:string|number, isCreate:boolean) {
+      const link = MatomoUrl.stringify({
+        ...MatomoUrl.urlParsed.value,
+        module: 'CoreHome',
+        action: 'index',
+      });
+      const hash = MatomoUrl.stringify({
+        ...MatomoUrl.hashParsed.value,
+        category: 'Goals_Goals',
+        subcategory: encodeURIComponent(goalId),
+      });
+      let successMessage = translate(isCreate ? 'Goals_GoalCreated' : 'Goals_GoalUpdated');
+      const reportLink = `<a href="?${link}#${hash}">[${translate('Goals_ViewGoalReport')}]</a>`;
+      successMessage = `<div class="notification-message">${successMessage} ${reportLink}</div>`;
```

Comment:
```suggestion
      successMessage = `${successMessage} ${reportLink}`;
```

In the UI test you can access the message using `.notification-body`, if that was the sole reason for this additional `div`.

Reply:
I've tested this and implemented this suggestion. Updated the UI test to compensate.

Reply:
Thanks for doing an update on this @nathangavin 🏆 

Cheers!